### DISCUSSION TITLE
Is everything in home/storage safe to upgrade termux?

### DISCUSSION THREAD

**Discussion by jxu** [termux/termux-app#2852] [Q&A] [answered]
Termux told me my Google play version is too old and I should backup termux files before installing a new version.

Is all the directories dcim, download, movies, music, pictures, shared in /data/data/com.termux/files/home/storage safe? It looks like all symlinks to internal android storage (/storage/emulated/0)

**Comment by agnostic-apollo** [comment, marked as answer]
Any data under `~/storage/external-1` will be deleted. Rest will remain after uninstall.

https://wiki.termux.com/wiki/Internal_and_external_storage

https://wiki.termux.com/wiki/Termux-setup-storage

**Reply by jxu** [reply]
Thanks. I had some throwaway files under ~/storage directly but everything else appears to be symlinks. 

Also I don't have external-1 directory.
